### DISCUSSION TITLE
Add option to manage Nvidia display mode

### DISCUSSION THREAD

**Discussion by Aj1234119** [seerge/g-helper#3411] [Ideas]
Hello,

On my system, a 2024 Zephyrus G16 with RTX 4090, I have the option to enable "Optimus" or "NVIDIA GPU only" through the NVIDIA control panel (see picture below). "NVIDIA GPU only" effectively routes the GPU directly to the display, providing better performance than Optimus and allowing for the use of Gsync without having to restart the laptop in Ultimate Mode. There is also an automatic option, but in my experience this option often fails to recognize proper situations to enable "NVIDIA GPU only" so there is value in adjusting the setting manually.

Would it be possible to integrate this choice of NVIDIA display mode into Ghelper (e.g. have a drop-down which allows for selection between automatic, Optimus, and NVIDIA GPU only, or an option to have Standard and Optimized Ghelper GPU modes default to one of these 3 options) so users can quickly switch between modes without having to open the NVIDIA control panel?

Thank you!
<img width="812" alt="display mode setting" src="https://github.com/user-attachments/assets/e5e08002-b58c-4b16-89d5-371c8d71ed10">

**Comment by savi4587** [comment]
![Screenshot 2024-11-25 202902](https://github.com/user-attachments/assets/1b49d17b-f20c-4e58-b499-750d6ae3c450)
Hi, Nvidia itself has a tray icon menu for switching between Optimus and Nvidia GPU only, no need for him to integrate it into the app as this is as easy to work with. just check the option " show dynamic display mode tray icon" and switch from the tray icon menu

**Comment by seerge** [comment]
@Aj1234119 hello, unfortunately that’s not possible because there is no API from Nvidia to do that.

But as other user mentioned you can add a tray icon that will let you switch fast between those modes.
